platforms/common/include/px4_platform_common/px4_work_queue/WorkItem.hpp

```diff
@@ -57,11 +57,20 @@ class WorkItem : public ListNode<WorkItem *>, public IntrusiveQueueNode<WorkItem
 	WorkItem(WorkItem &&) = delete;
 	WorkItem &operator=(WorkItem &&) = delete;
 
-	inline void ScheduleNow()
+	/**
+	 * Schedule the WorkItem to run immediately.
+	 *
+	 * @return true if the WorkItem was scheduled successfully, false otherwise.
+	 */
+	inline bool ScheduleNow()
 	{
 		if (_wq != nullptr) {
-			_wq->Add(this);
+			if ((_configured_interval_us == 0) || (hrt_elapsed_time_atomic(&_time_last_run) >= _configured_interval_us)) {
```

Comment:
The name suggests otherwise, but using `hrt_elapsed_time_atomic` is not safe here. The only safe use-case is if the following hold:
- a timestamp that is updated from IRQ context only (or with IRQ's disabled)
- and used on NuttX